x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/Engine.java

```diff
@@ -43,26 +43,24 @@ public class Engine implements Writeable, ToXContentObject {
     public static final String ENGINE_ALIAS_PREFIX = "engine-";
     private final String name;
     private final String[] indices;
-    private long updatedAtMillis = System.currentTimeMillis();
+    private static final ConstructingObjectParser<Engine, String> PARSER = new ConstructingObjectParser<>(
```

Comment:
Ah, thanks, I see where you were going. Apologies, I didn't realize this was what you were thinking in our previous discussion, but I can see how it would work here. 

TBH, though, I kind of like the mutable version a little better since it means we don't have to add nulls to every constructor.